tensorboard/components/tf_ng_tensorboard/header/containers/header.component.ts

```diff
@@ -13,29 +13,40 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 import {Component} from '@angular/core';
+import {MatTabChangeEvent} from '@angular/material/tabs';
+import {MatSelectChange} from '@angular/material/select';
 import {Store, select, createSelector} from '@ngrx/store';
 import {combineLatest, of} from 'rxjs';
 
 import {State, getActivePlugin, getPlugins} from '../../core/core.reducers';
 import {changePlugin} from '../../core/core.actions';
 
+import {PluginMetadata, PluginId} from '../../types/api';
+
 /** @typehack */ import * as _typeHackRxjs from 'rxjs';
 
+interface UiPluginMetadata extends PluginMetadata {
+  id: PluginId;
+}
+
 const selectPlugins = createSelector(
   getPlugins,
-  (listing) =>
+  (listing): UiPluginMetadata[] =>
```

Comment:
Do I understand correctly that these are strongly typed, even though the
types are inferred, so this is `Selector<BigState, UiPluginMetadata[]>`
or something?

Reply:
Yup, that's correct. `BigState` -> `PluginMetadata[]` -> `UiPluginMetadata[]`.